src/Npgsql/VolatileResourceManager.cs

```diff
@@ -138,6 +143,7 @@ public void Commit(Enlistment enlistment)
 
             try
             {
+                _isRMCall.Value = true;
```

Comment:
Wouldn't it be easier to simply record Thread `Thread.CurrentThread.ManagedThreadId` on the RM and then check in WaitIfRequired if the current thread ID matches that? Seems simpler than a ThreadLocal unless I'm missing something...